File: java-project/StudentSystem.java
// StudentSystem.java

// Placeholder code for Student Management System
import java.util.HashMap;
import java.util.Map;

public class StudentSystem {

    static class Student {
        String name;
        int rollNo;
        int age;

        public Student(String name, int rollNo, int age) {
            this.name = name;
            this.rollNo = rollNo;
            this.age = age;
        }

        @Override
        public String toString() {
            return "Name: " + name + ", Roll No: " + rollNo + ", Age: " + age;
        }
    }

    static class StudentSystemManager {
        Map<Integer, Student> students = new HashMap<>();

        public void addStudent(String name, int rollNo, int age) {
            if (students.containsKey(rollNo)) {
                System.out.println("Student with roll number " + rollNo + " already exists.");
            } else {
                students.put(rollNo, new Student(name, rollNo, age));
                System.out.println("Student " + name + " added successfully.");
            }
        }

        public void updateStudent(int rollNo, String name, Integer age) {
            if (students.containsKey(rollNo)) {
                Student student = students.get(rollNo);
                if (name != null) {
                    student.name = name;
                }
                if (age != null) {
                    student.age = age;
                }
                System.out.println("Student with roll number " + rollNo + " updated.");
            } else {
                System.out.println("No student found with roll number " + rollNo);
            }
        }

        public void deleteStudent(int rollNo) {
            if (students.containsKey(rollNo)) {
                students.remove(rollNo);
                System.out.println("Student with roll number " + rollNo + " deleted.");
            } else {
                System.out.println("No student found with roll number " + rollNo);
            }
        }

        public void displayStudents() {
            if (students.isEmpty()) {
                System.out.println("No students available.");
            } else {
                System.out.println("List of students:");
                for (Student student : students.values()) {
                    System.out.println(student);
                }
            }
        }
    }

    public static void main(String[] args) {
        System.out.println("Welcome to the Student Management System!");
        StudentSystemManager system = new StudentSystemManager();
        system.addStudent("Alice", 101, 20);
        system.addStudent("Bob", 102, 22);
        system.updateStudent(101, null, 21);
        system.displayStudents();
        system.deleteStudent(102);
        system.displayStudents();
    }
}

File: python-project/student_system.py
# student_system.py

# Placeholder code for Student Management System
class Student:
    def __init__(self, name, roll_no, age):
        self.name = name
        self.roll_no = roll_no
        self.age = age

    def __str__(self):
        return f"Name: {self.name}, Roll No: {self.roll_no}, Age: {self.age}"

class StudentSystem:
    def __init__(self):
        self.students = {}

    def add_student(self, name, roll_no, age):
        if roll_no in self.students:
            print(f"Student with roll number {roll_no} already exists.")
        else:
            self.students[roll_no] = Student(name, roll_no, age)
            print(f"Student {name} added successfully.")

    def update_student(self, roll_no, name=None, age=None):
        if roll_no in self.students:
            student = self.students[roll_no]
            if name:
                student.name = name
            if age:
                student.age = age
            print(f"Student with roll number {roll_no} updated.")
        else:
            print(f"No student found with roll number {roll_no}.")

    def delete_student(self, roll_no):
        if roll_no in self.students:
            del self.students[roll_no]
            print(f"Student with roll number {roll_no} deleted.")
        else:
            print(f"No student found with roll number {roll_no}.")

    def display_students(self):
        if not self.students:
            print("No students available.")
        else:
            print("List of students:")
            for student in self.students.values():
                print(student)

# Placeholder main function
def main():
    print("Welcome to the Student Management System!")
    system = StudentSystem()
    system.add_student("Alice", 101, 20)
    system.add_student("Bob", 102, 22)
    system.update_student(101, age=21)
    system.display_students()
    system.delete_student(102)
    system.display_students()

if __name__ == "__main__":
    main()
